linera-base/src/identifiers.rs

```diff
@@ -819,6 +836,140 @@ impl<A> ApplicationId<A> {
     }
 }
 
+// Cannot use #[derive(Clone)] because it requires `A: Clone`.
+impl<A> Clone for BlobApplicationId<A> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+impl<A> Copy for BlobApplicationId<A> {}
+
+impl<A: PartialEq> PartialEq for BlobApplicationId<A> {
+    fn eq(&self, other: &Self) -> bool {
+        self.application_description_hash == other.application_description_hash
+            && self.bytecode_id == other.bytecode_id
+    }
+}
+
+impl<A: Eq> Eq for BlobApplicationId<A> {}
+
+impl<A: PartialOrd> PartialOrd for BlobApplicationId<A> {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        (self.application_description_hash, self.bytecode_id)
+            .partial_cmp(&(other.application_description_hash, other.bytecode_id))
+    }
+}
+
+impl<A: Ord> Ord for BlobApplicationId<A> {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        (self.application_description_hash, self.bytecode_id)
```

Comment:
If we destructure in these methods, the compiler will warn us in case we add any additional fields.